packages/amplify-provider-awscloudformation/src/pre-push-cfn-processor/cfn-pre-processor.ts

```diff
@@ -24,3 +39,176 @@ export async function preProcessCFNTemplate(filePath: string): Promise<string> {
   await writeCFNTemplate(cfnTemplate, newPath, { templateFormat });
   return newPath;
 }
+
+//get data from custom polcies file and write custom policies to CFN template
+export async function writeCustomPoliciesToCFNTemplate(
+  resourceName: string,
+  service: string,
+  resourceDir: string,
+  cfnFile: string,
+  category: string,
+  filePath: string
+) {
+  if (!(service === 'Lambda' || service === 'ElasticContainer')) {
+    return;
+  }
+  const { templateFormat, cfnTemplate } = await readCFNTemplate(path.join(resourceDir, cfnFile));
+  const customPolicies = stateManager.getCustomPolicies(category, resourceName);
+  if (!validateExistCustomPolicies(customPolicies)) {
+    if (cfnTemplate.Resources.CustomLambdaExecutionPolicy) {
+      delete cfnTemplate.Resources.CustomLambdaExecutionPolicy;
+      await writeCFNTemplate(cfnTemplate, filePath, { templateFormat });
+    }
+    if (cfnTemplate.Resources.CustomExecutionPolicyForContainer) {
+      delete cfnTemplate.Resources.CustomExecutionPolicyForContainer;
+      await writeCFNTemplate(cfnTemplate, filePath, { templateFormat });
+    }
+    return;
+  }
+  await validateCustomPoliciesSchema(customPolicies, category, resourceName);
+
+  await addCustomPoliciesToCFNTemplate(service, category, customPolicies, cfnTemplate, filePath, resourceName, {templateFormat} );
+
+}
+
+//merge the custom IAM polciies to CFN template for lambda and API container
+
+async function addCustomPoliciesToCFNTemplate(
+  service: string,
+  category: string,
+  customPolicies: CustomIAMPolicies,
+  cfnTemplate: Template,
+  filePath: string,
+  resourceName: string,
+  {templateFormat}: any
```

Comment:
Can we Define a type for `templateformat `of this as `JSON|YAML`. But I think we already have since we are using it in above in `writeCFNtemplate` function